shell/platform/android/io/flutter/embedding/engine/loader/ResourceExtractor.java

```diff
@@ -154,9 +115,50 @@ ResourceExtractor start() {
       Log.e(
           TAG, "Attempted to start resource extraction while another extraction was in progress.");
     }
+
     mExtractTask =
-        new ExtractTask(mDataDirPath, mResources, mPackageName, mPackageManager, mAssetManager);
-    mExtractTask.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
+        new FutureTask(
+            new Runnable() {
+              @Override
+              public void run() {
+                final File dataDir = new File(mDataDirPath);
+
+                final String timestamp = checkTimestamp(dataDir, mPackageManager, mPackageName);
+                if (timestamp == null) {
+                  return;
+                }
+
+                deleteFiles(mDataDirPath, mResources);
+
+                if (!extractAPK(dataDir)) {
+                  return;
+                }
+
+                if (timestamp != null) {
+                  try {
+                    new File(dataDir, timestamp).createNewFile();
+                  } catch (IOException e) {
+                    Log.w(TAG, "Failed to write resource timestamp");
+                  }
+                }
+
+                return;
+              }
+            },
+            null);
+
+    ThreadPoolExecutor executor =
+        new ThreadPoolExecutor(
+            1 /* CORE POOL SIZE */,
```

Comment:
Could you clarify what led to these settings for the thread pool?

Reply:
Yeah, I used the same settings that were being used under the hood before -- see [here](https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/os/AsyncTask.java#211).

Reply:
For future maintainability, you can actually directly link the source for this here in a comment, that way we can easily find the thought process behind it (and have context for changing it if we would like to).

Reply:
+1 something like: "Use the same default values from the deprecated android.os.AsyncTask API." 

Sometimes, these comments are useful to folks in g3 for example.